File: answers/Fab-Gaurav/Day 2/Answer_1.py
#Program to check if the number "n" is Emirp or not

n=int(input("Enter the number : "))
def prime(t,i):
    if(i<t):
        if(t%i!=0):
            return prime(t,i=i+1)
        else:
            return 0
    else:
        return 1

p=prime(n,2)
if p==1:
    s=0
    c=n
    while(c>0):
        r=c%10
        s=s*10+r
        c=c//10
    p=prime(s,2)
    if p==1:
        print("%d is an Emirp number"%n)
    else:
        print("%d is not an Emirp number"%n)
else:
    print("%d is not an Emirp number"%n)

File: answers/Shobhit403/DAY 2/Answer1.cpp
//Program to check if the number is an Emirp Number
#include <iostream> 
using namespace std; 
bool isPrime(int n) 
{ 
    if (n <= 1) 
        return false; 
    for (int i = 2; i < n; i++) 
        if (n % i == 0) 
            return false; 
    return true; 
} 
bool isEmirp(int n) 
{ 
    if (isPrime(n) == false) 
        return false; 
    int rev = 0; 
    while (n != 0) { 
        int d = n % 10; 
        rev = rev * 10 + d; 
        n /= 10; 
    } 
    return isPrime(rev); 
} 
int main() 
{ 
    int n = 13;
    if (isEmirp(n) == true) 
        cout << n <<" IS AN EMIRP NUMBER"; 
    else
        cout << n <<" IS NOT AN EMIRP NUMBER"; 
}
